Code/GraphMol/Chirality.cpp

```diff
@@ -1791,6 +1791,87 @@ void assignChiralTypesFrom3D(ROMol &mol, int confId, bool replaceExistingTags) {
   }
 }
 
+void assignChiralTypesFromMolParity(ROMol &mol, bool replaceExistingTags) {
+  // if the molecule already has stereochemistry
+  // perceived, remove the flags that indicate
+  // this... what we're about to do will require
+  // that we go again.
+  static const std::vector<Atom::ChiralType> chiralTypeVect {
+    Atom::CHI_TETRAHEDRAL_CW,
+    Atom::CHI_TETRAHEDRAL_CCW
+  };
+  if (mol.hasProp(common_properties::_StereochemDone)) {
+    mol.clearProp(common_properties::_StereochemDone);
+  }
+  // Atom-based parity
+  //
+  // Parity 1 (CW)        Parity 2 (CCW)
+  //     3   1                3   2
+  //      \ /                  \ /
+  //       |                    |
+  //       2                    1
+  //
+  for (auto atomIt = mol.beginAtoms(); atomIt != mol.endAtoms(); ++atomIt) {
+    Atom *atom = *atomIt;
+    // if we aren't replacing existing tags and the atom is already tagged,
+    // punt:
+    if (!replaceExistingTags && atom->getChiralTag() != Atom::CHI_UNSPECIFIED)
+      continue;
+    int parity = 0;
+    atom->getPropIfPresent(common_properties::molParity, parity);
+    if (parity <= 0 || parity > 2 || atom->getDegree() < 3) {
+      atom->setChiralTag(Atom::CHI_UNSPECIFIED);
+      continue;
+    }
+    // if we are here, parity was 1 (CW) or 2 (CCW)
+    // now we set parity 0 to be CW and 1 to be CCW
+    --parity;
+    RDKit::ROMol::OBOND_ITER_PAIR nbrBonds = mol.getAtomBonds(atom);
+    std::vector<const Bond *> nbrBondVect;
+    std::transform(nbrBonds.first, nbrBonds.second,
+      std::back_inserter(nbrBondVect), [mol](const ROMol::edge_descriptor &e) {
+        return mol[e];
+      });
+    std::sort(nbrBondVect.begin(), nbrBondVect.end(), [](const Bond *a, const Bond *b) {
+      return (a->getIdx() < b->getIdx());
+    });
+    unsigned int atomIdx = atom->getIdx();
+    std::vector<unsigned int> nbrAtomIdxVectSortedByBond;
```

Comment:
same suggestion here:
```suggestion
    std::vector<unsigned int> nbrAtomIdxVectSortedByBond;
    nbrAtomIdxVectSortedByBond.reserve(nbrBondVect.size());
```

Reply:
Ditto.